### PR TITLE
rust: proc-macro should be ignored in transitive dependencies

### PR DESCRIPTION
**Pull request by xclaesse**
Fixes: #12459

### CODE REVIEW

**Comment by xclaesse** [comment]
We don't currently have Rust cross compilation CI so the unit test does not really catch the issue ATM, but I verified manually that `libpm.so` was passed to footest link command and this PR removed it.

File: test cases/rust/20 transitive dependencies/foo.c [hunk 1]
```diff
@@ -0,0 +1,6 @@
+int foo_rs(void);
```
Comment (gfxstrand) [inline test cases/rust/20 transitive dependencies/foo.c:1]: Not that it actually matters but the signedness is different between your C prototype and the Rust. Maybe use `unsigned` instead? Or make the rust `i32`?

**Comment by gfxstrand** [comment]
Other than the one comment above, everything looks good to my uneducated (about meson) eye. I've also tested it with my Mesa nak branch and it fixes the bug. 32-bit builds seem to work okay now.

File: test cases/rust/20 transitive dependencies/foo.c (hunk 1 above)
Comment (xclaesse) [reply test cases/rust/20 transitive dependencies/foo.c]: indeed, fixed.

**Review by dcbaker — APPROVED** [review]

**Comment by eli-schwartz** [comment]
I really think we need to add rust cross compilation to the CI if we're going to keep poking at this code. What's the status of doing so??

**Comment by dcbaker** [comment]
I will have to add it with my series to build subprojects for both machines if not sooner, so... during the next release cycle at the latest?

**Comment by eli-schwartz** [comment]
> I made a PR to add Windows cross compilation and it exploded in million pieces: #11990. I'll try with Linux aarch64 as an easier first step.

The CI image builder now has the aarch64 bits installed in it, and deployed.

**Pull request merged**